src/devices/bus/qbus/dvk_mx.h

```diff
@@ -0,0 +1,170 @@
+// license:BSD-3-Clause
+// copyright-holders:Sergey Svishchev
+/***************************************************************************
+
+	DVK "MX" floppy controller (decimal ID 3.057.122)
+
+***************************************************************************/
+
+#ifndef MAME_BUS_QBUS_DVK_MX_H
+#define MAME_BUS_QBUS_DVK_MX_H
+
+#pragma once
+
+#include "qbus.h"
+
+#include "machine/fdc_pll.h"
+#include "machine/pdp11.h"
+#include "imagedev/floppy.h"
+
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> dvk_mx_device
+
+class dvk_mx_device : public device_t, public device_qbus_card_interface
+{
+public:
+	dvk_mx_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	uint16_t read(offs_t offset);
+	void write(offs_t offset, uint16_t data);
+
```

Comment:
Looks like that fdc can't interrupt the main cpu.  If it's the case there is a somewhat better structure than predict/commit (like the wd/upd/amiga do), it's catchup on access (like diskii/iwm/swim1-2).  But it's good anyway.

Reply:
I have another floppy code in the works, will look into that.